### DISCUSSION TITLE
Option to remove search bar from messages in topic

### DISCUSSION THREAD

**Discussion by vSivarajah93** [provectus/kafka-ui#4111] [Q&A] [answered]
Hello! We've deployed a Kafka-UI instance where all topic data is masked.

Despite this, using the search function, I can still locate entries by entering the correct value although it shows as masked.

<img width="1933" alt="Skjermbilde 2023-08-11 kl  17 12 41" src="https://github.com/provectus/kafka-ui/assets/135104283/dfe8410c-9485-47c6-ad11-e0fc99f477a1">
in the screenshot above i am deliberatly searching for the value: "HmacSHA256" and it gives 4 entries 

Is there an option to disable this search functionality? We introduced this UI to allow developers an easy way to verify if messages have been received in a topic. Our Kafka cluster is running behind a private endpoint, making it inherently difficult to access Kafka by default.

**Comment by Haarolean** [comment, marked as answer]
Hi, regarding a data mask bug I've raised a new issue #4115, we'll fix that the next week.

Currently it's possible to prevent users from viewing messages altogether with [RBAC](https://docs.kafka-ui.provectus.io/configuration/rbac-role-based-access-control).
